### PR TITLE
Fixed `get_list()` to not show incompatible scan type options

### PR DESCRIPTION
**Pull request by OpenNetSurfer**
- `String` or `AOB` value types when selected will not show incompatible scan types as a option

### CODE REVIEW

File: libpince/typedefs.py [hunk 1]
```diff
@@ -333,31 +333,42 @@ class SCAN_TYPE:
...
+                    SCAN_TYPE.LESS,
+                    SCAN_TYPE.MORE,
+                    SCAN_TYPE.BETWEEN,
+                    SCAN_TYPE.CHANGED,
+                    SCAN_TYPE.UNCHANGED,
+                ]
+            else:
+                list = [
+                    SCAN_TYPE.EXACT,
+                    SCAN_TYPE.CHANGED,
```
Comment (brkzlr) [inline libpince/typedefs.py:369]: Strings and AOB support only `EXACT` in this scenario, not `CHANGED` or `UNCHANGED`

**Review by brkzlr — CHANGES_REQUESTED** [review]

File: libpince/typedefs.py (hunk 1 above)
Comment (brkzlr) [reply libpince/typedefs.py:369]: Also please make sure you use the Black formatter. Your previous PR was unformatted.

Comment (OpenNetSurfer) [reply libpince/typedefs.py:369]: > Strings and AOB support only EXACT in this scenario, not CHANGED or UNCHANGED

Missed that.

> Also please make sure you use the Black formatter. Your previous PR was unformatted.

Could you explain what is meant by black formatter? I did read in `CONTRIBUTING.md` but didn't understand.

Comment (brkzlr) [reply libpince/typedefs.py:369]: We're using this formatter for PINCE: https://github.com/psf/black

You have to use that formatter before any PR to make sure that the code style is consistent.

Comment (OpenNetSurfer) [reply libpince/typedefs.py]: Fixed it. But regarding black formatter, it showed everything was fine and nothing changed.

Comment (brkzlr) [reply libpince/typedefs.py]: This is still wrong.

`STRING` does not accept `CHANGED` or `UNCHANGED` like I've said before.
Removing `UNKNOWN` from `AOB` was also unnecessary.

Revert your last commit and just remove the `CHANGED` and `UNCHANGED` value from **BOTH** `STRING` and `AOB` like previously stated.

Comment (OpenNetSurfer) [reply libpince/typedefs.py]: Understood

**Comment by brkzlr** [comment]
Apologies, I came back to this late because you force-pushed the last commit, keeping the commit message same so GitHub didn't tell me you changed anything. Just noticed it now when I manually checked your last commit from your fork.

I'll close this PR and fix the bug myself as you either keep ignoring what I'm asking you to do or not fully reading my review comments as you now removed `CHANGED`/`UNCHANGED` from `STRING` like I asked you too but you still didn't restore `UNKNOWN` for `AOB` like I asked and even removed it from `STRING` now, a valid scan type for both.

Sorry but doing constant back and forth for a very simple change request is not great.

**Comment by OpenNetSurfer** [comment]
I am sorry, I couldn't understand well. Will not happen again

**Pull request closed without merging**